GVFS/GVFS.Common/Prefetch/BlobPrefetcher.cs

```diff
@@ -229,6 +233,17 @@ public virtual void Prefetch(string branchOrCommit, bool isBranch)
                 }
             }
 
+            // TODO: make this a separate public method so that the caller can exit even earlier. Right now we're still wasting time
+            //       on auth and config even when we know there will be nothing to do

```

Comment:
Would this be easier if the data was abstracted into a `PrefetchRunData` class that wraps the `FileBasedDictionary` so it could be created and used outside this class?

Reply:
I don't think a new wrapper class will change anything. The caller is already the one instantiating the class, so that's not the issue. Just need to put this code in a public method, is all.